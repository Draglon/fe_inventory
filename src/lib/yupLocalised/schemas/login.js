import * as yup from "yup";
import {
  MAX_PASSWORD_LENGTH,
  MIN_PASSWORD_LENGTH,
} from "@/lib/constants";

export default yup.object().shape({
  email: yup.string().email().required(),
  password: yup.string().min(MIN_PASSWORD_LENGTH).max(MAX_PASSWORD_LENGTH).required(),
});
