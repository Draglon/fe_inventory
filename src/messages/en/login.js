const login = {
  "Login": {
    "layout": {
      "title": "Login",
      "description": "Login - description",
    },
    "title": "Login",
  },
};

export default login;