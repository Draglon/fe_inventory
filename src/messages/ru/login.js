const login = {
  "Login": {
    "layout": {
      "title": "Авторизоваться",
      "description": "Авторизоваться - описание",
    },
    "title": "Авторизоваться",
  },
};

export default login;