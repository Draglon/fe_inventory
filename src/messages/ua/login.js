const login = {
  "Login": {
    "layout": {
      "title": "Вхід",
      "description": "Вхід - опис",
    },
    "title": "Вхід",
  },
};

export default login;