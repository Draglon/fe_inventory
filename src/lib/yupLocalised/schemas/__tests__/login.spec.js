import login from "../login";

describe("login validation schema", () => {
  it("matches snapshot", () => {
    expect(login.describe()).toMatchSnapshot();
  });
});
