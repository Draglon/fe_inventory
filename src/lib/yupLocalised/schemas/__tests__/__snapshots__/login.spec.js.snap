// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`login validation schema matches snapshot 1`] = `
{
  "default": {
    "email": undefined,
    "password": undefined,
  },
  "fields": {
    "email": {
      "default": undefined,
      "label": undefined,
      "meta": undefined,
      "notOneOf": [],
      "nullable": false,
      "oneOf": [],
      "optional": false,
      "tests": [
        {
          "name": "email",
          "params": {
            "regex": /\\^\\[a-zA-Z0-9\\.!#\\$%&'\\*\\+\\\\/=\\?\\^_\`\\{\\|\\}~-\\]\\+@\\[a-zA-Z0-9\\]\\(\\?:\\[a-zA-Z0-9-\\]\\{0,61\\}\\[a-zA-Z0-9\\]\\)\\?\\(\\?:\\\\\\.\\[a-zA-Z0-9\\]\\(\\?:\\[a-zA-Z0-9-\\]\\{0,61\\}\\[a-zA-Z0-9\\]\\)\\?\\)\\*\\$/,
          },
        },
        {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
    "password": {
      "default": undefined,
      "label": undefined,
      "meta": undefined,
      "notOneOf": [],
      "nullable": false,
      "oneOf": [],
      "optional": false,
      "tests": [
        {
          "name": "min",
          "params": {
            "min": 6,
          },
        },
        {
          "name": "max",
          "params": {
            "max": 72,
          },
        },
        {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
  },
  "label": undefined,
  "meta": undefined,
  "notOneOf": [],
  "nullable": false,
  "oneOf": [],
  "optional": true,
  "tests": [],
  "type": "object",
}
`;
